Swann

OshidaBCF opened the pull request.

KevinTerrisson committed:
Gird

KevinTerrisson committed:
Create Board.meta

KevinTerrisson committed:
test

KevinTerrisson committed:
Update Test.unity

KevinTerrisson committed:
turn beggin

OshidaBCF committed:
Added Grid

TheoBrs commented on Assets/Scripts/Board/Mouvment.cs:
Renommer en "Movement"

OshidaBCF committed:
Merge branch 'TheoBrs:main' into Swann

TheoBrs reviewed (changes requested):
- Enlever tout le code mort (les "using", fonctions... Tout ce qui est inutilisé)
- Eviter le plus possible les variables publiques -> Utiliser des get/set, récupérer une variable privée à travers une fonction, mettre des variables privées en [SerializeField], ect...
- Revoir les conventions de nommage pour toutes les variables (se servir du fichier texte envoyé par Lilian)
- Merge avec le main
- Rendre les cases plus visible (ex: faire un contour à chaque case)

OshidaBCF committed:
Fixed PR Requests

OshidaBCF committed:
"finished" grid

TheoBrs commented on Assets/Scripts/Board/GridElement.cs:
```diff
@@ -0,0 +1,16 @@
+using UnityEngine;
+
+namespace Board
```
Pourquoi namespace?

TheoBrs commented on Assets/Scripts/Board/Grid.cs:
```diff
@@ -0,0 +1,35 @@
+using UnityEngine;
+
+namespace Board
```
Namespace board à enlever

TheoBrs commented on Assets/Scripts/Board/Grid.cs:
```diff
@@ -0,0 +1,35 @@
+
+namespace Board
+{
+    public class Grid : MonoBehaviour
```
Modifier le nom "Grid" pour un nom plus explicite

TheoBrs commented on Assets/Scripts/Character/Entity.cs:
```diff
@@ -1,27 +1,30 @@
+
+    private void Awake()
+    {
+        _grid = GameObject.Find("WorldPlane");
```
Pas de Find car pas optimisé et ça peut casser facilement

Firzus commented on .editorconfig:
Enlever

TheoBrs commented on Assets/Scripts/Gameplay/Turn.cs:
Script inutile

OshidaBCF replied:
Pour l'instant 

Ask kevin

TheoBrs commented on Assets/Scripts/Gameplay/Ability.cs:
```diff
@@ -19,4 +19,6 @@ public int RoundsBeforeReuse
         set { _roundsBeforeReuse = value; }
     }
+
+    public Ability() {}
```
Ne sert à rien

TheoBrs commented on Assets/Scripts/Character/Player.cs:
```diff
@@ -14,15 +17,32 @@ public void Start()
+            Damage = 4,
+            Cost = 2
+        };
+        // why the "f" in front of ability ? Fake ?
```
Pas besoin de mettre un commentaire pour ça, ça sert à rien

TheoBrs reviewed (changes requested):
- Il ne faut pas que les variables soient publiques, elles doivent être privée/protected
- Pas de déplacement

Firzus commented on Assets/Prefabs/System/GridCell.prefab:
Grid Cell pas dans le bon dossier

Firzus commented on Assets/Scripts/Board/Coord.cs:
```diff
@@ -0,0 +1,27 @@
+       
+        public Coord(int x, int y)
+        {
+            this.x = x;
```
this inutile

OshidaBCF replied:
"x" = "x" est ambigu

Firzus commented on Assets/Scripts/Board/Grid.cs:
```diff
@@ -0,0 +1,35 @@
+                }
+            }
+        }
+        void Update()
```
code mort

Firzus commented on Assets/Scripts/Board/Movement.cs:
```diff
@@ -0,0 +1,31 @@
+    public Vector3 newPosition;
+    public float movementSpeed;
+
+    // Start is called before the first frame update
```
enlever les commentaires inutiles par défaut

Firzus commented on Assets/Scripts/Board/Movement.cs:
```diff
@@ -0,0 +1,31 @@
+    }
+
+    // Update is called once per frame
+    void Update()
```
la physique dans un fixed update

Firzus commented on Assets/Scripts/Character/Entity.cs:
```diff
@@ -1,27 +1,30 @@
     [HideInInspector] public Stat Attack;
     [HideInInspector] public Stat Defense;
 
+    public Coord _currentPos; // Public for Testing
```
comme tout le reste

Firzus commented on Assets/Scripts/Character/Entity.cs:
```diff
@@ -1,27 +1,30 @@
 
     protected bool _invincible = false;
 
+    GameObject _grid;
```
pourquoi en public ?

OshidaBCF replied:
C'est pas public

Firzus commented on Assets/Scripts/Character/Entity.cs:
```diff
@@ -56,6 +59,24 @@ private void IsDead()
         }
     }
 
+    public void Move(Coord to)
```
un nom plus explicite que "to"

Firzus commented on Assets/Scripts/Character/Entity.cs:
```diff
@@ -56,6 +59,24 @@ private void IsDead()
+    {
+        GridElement gridElement = _grid.GetComponent<Board.Grid>().elements[to.x, to.y];
+
+        Vector3 newPosition = gridElement.gridElement.transform.position;
```
gridElement.gridElement ??????????

Firzus commented on Assets/Scripts/Character/Player.cs:
```diff
@@ -95,6 +115,15 @@ protected void CastFriendAbility3()
 
+    void Update()
+    {
+        if (Input.GetKey(KeyCode.E))
```
exporter dans une autre fonction

OshidaBCF committed:
Changes for PR

OshidaBCF committed:
Fix PR part 2

OshidaBCF committed:
Removed most public stuff

OshidaBCF committed:
Removed Unused Package

TheoBrs commented on Assets/Scripts/Character/Entity.cs:
```diff
@@ -2,26 +2,26 @@
     [Header("HP and AP settings")]
-    public Stat MaxHP;
-    public Stat MaxAP;
+    [SerializeField] Stat MaxHP;
```
Pas besoin de mettre [SerializeField] sur ces deux lignes ici, le Header fait la même chose plus proprement

OshidaBCF replied:
J'ignorai ça, on ma dit de tout mettre en privé

TheoBrs commented on Assets/Scripts/Character/Entity.cs:
```diff
@@ -68,4 +86,14 @@ public void BattleEnd()
         _currentAP = MaxAP.GetValue();
     }
+
+    public Stat GetMaxHP()
```
J'avais déjà fait une fonction pour récupérer les valeurs de MaxHP et MaxAP, il faut pas les modifier

TheoBrs commented on Assets/Scripts/Character/Player.cs:
```diff
@@ -1,28 +1,62 @@
 
     protected override void AbilitiesInitialization()
     {
-        _ability1.Damage = 2;
```
Encore une fois ça sert à rien de modifier ce que j'ai déjà fait, il faut que ça revienne comme avant pour les abilities

OshidaBCF replied:
les variable ne sont pas instancier ducoup crash

J'ai annulé mes changement mais pour l'instant ces ligne seront en commentaire

TheoBrs commented on Assets/Scripts/Gameplay/Turn.cs:
Fonction inutile toujours

TheoBrs commented on Assets/Scripts/Character/Entity.cs:
```diff
@@ -2,26 +2,26 @@
 
 public abstract class Entity: MonoBehaviour
 {
-    public struct Coord
```
Pourquoi avoir supprimé ça?

OshidaBCF replied:
Unused + on a une autre structure Coord plus complete

TheoBrs reviewed (changes requested):
Ne modifie pas ce qui a déjà été fait. Si vraiment tu as besoin tu peux demander, mais normalement il faut pas toucher

OshidaBCF committed:
More PR Fix stuff

OshidaBCF committed:
Finished Grid Movement

I don't know yet how to grab materials from assets via code, so for now the two mats are serialied field

OshidaBCF committed:
Added thingy from discord
